### ISSUE TITLE
_fused_sdp_choice returns wrong backend in pt2

### ISSUE THREAD

**Issue by voznesenskym** [pytorch/pytorch#113524]
### 🐛 Describe the bug

TORCH_LOGS=+dynamo PYTORCH_TEST_WITH_DYNAMO=1 python -m pytest  test/test_transformers.py -k test_fused_sdp_choice_cpu_type_dense_dropout_0_0 -x

This used to graph break before https://github.com/pytorch/pytorch/pull/113512 on the fact that it returned a scalar value/enum. Now, it returns the wrong one. This never worked, but was hidden by graph breaks. 

### Versions

After https://github.com/pytorch/pytorch/pull/113512

**Comment by cpuhrsch** [comment]
@voznesenskym - Should we tag this as an issue for SDPA or an issue for torch.compile?

**Comment by cpuhrsch** [comment]
@voznesenskym This seems like a critical bug. Should we mark this as high priority?

**Comment by drisspg** [comment]
I don't think this is actually critical, we dont' use fused_sdp_choice when calling SDPA since this broke compile way back when. _fused_sdp_choice is what I wanted to use to standardize the logic between backends but now we do manual call stubs with the dtpe: https://github.com/pytorch/pytorch/blob/edd967fe78bca74dd35ef00830e03bd24153f796/aten/src/ATen/native/transformers/attention.cpp#L640

And use _fused_sdp_choice for testing since it ends up calling the same c++ code
